packages/react-utilities/src/hooks/useControllableState.test.ts

```diff
@@ -0,0 +1,140 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useControllableState } from './useControllableState';
+
+describe('useControllableState', () => {
+  afterEach(jest.resetAllMocks);
+
+  it('respects controlled state', () => {
+    let state: boolean;
+    let defaultState: boolean | undefined;
+    const initialState = false;
+
+    state = true;
+    const { result, rerender } = renderHook(() => useControllableState({ state, defaultState, initialState }));
+    expect(result.current[0]).toBe(state);
+
+    state = false;
+    rerender();
+    expect(result.current[0]).toBe(state);
+
+    state = false;
+    defaultState = true;
+    rerender();
+    expect(result.current[0]).toBe(state);
```

Comment:
Can we use [built-in API](https://react-hooks-testing-library.com/usage/basic-hooks#providing-props) in `rerender()` instead of mutations? 

For example, it will look like this:

```tsx
    rerender({ state: false, defaultState: true });
    expect(result.current[0]).toBe(false);
```

Reply:
Did it for the first test, since it's overkill to redefine the entire rerender props for each rerender. As stated in [the docs](https://react-hooks-testing-library.com/usage/basic-hooks) mutating is fine for simple tests